app/models/manageiq/providers/cloud_automation_manager/configuration_manager/refresher.rb

```diff
@@ -4,8 +4,16 @@ module CloudAutomationManager
     class ConfigurationManager::Refresher < ManageIQ::Providers::BaseManager::Refresher
       def parse_legacy_inventory(manager)
         manager.with_provider_connection do |connection|
-          data = collect_configuration_inventory(connection, manager.url)
+          # get templates
```

Comment:
Then the caller side can look like this:
```
data = {}

# get templates
template_url_str = ":30000/cam/api/v1/templates"
data[:templates] = collect_configuration_inventory(connection, manager.url, template_url_str)

# get configured systems
stack_url_str = ":30000/cam/api/v1/stacks"
data[:stacks] = collect_configuration_inventory(connection, manager.url, stack_url_str)

ConfigurationManager::RefreshParser.configuration_inv_to_hashes(data)
```

Reply:
Yeah good call - that is cleaner.

Reply:
`collect_configuration_inventory` is expected to return a hash with all of the collected data